main/src/ca/uwaterloo/flix/api/Bootstrap.scala

```diff
@@ -424,6 +425,55 @@ class Bootstrap(val projectPath: Path, apiKey: Option[String]) {
     }
   }
 
+  /**
+    * Sequentially deletes all `.class` files in the build directory if ALL files (recursively) in the build directory
+    * are `.class` files.
+    * Aborts if any other file was found.
+    */
+  def clean(): Result[Unit, BootstrapError] = {
+    val buildDir = getBuildDirectory(projectPath)
+    val root = Path.of("/").normalize()
+
+    if (buildDir == root) {
+      return Err(BootstrapError.FileError("unexpected build dir"))
+    }
```

Comment:
Are there any other special cases like this we can think of? We want to be paranoid here.